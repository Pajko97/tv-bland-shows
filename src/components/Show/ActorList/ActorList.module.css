.content_column {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  flex-basis: grow;
  width: 20em;
  width: 100%;
  padding: 2rem;
  padding-left: 0;
  margin-top: 2rem;
}

.column_item {
  display: flex;
  border-bottom: 1px solid black;
}
.item_text {
  display: flex;
}

.column_item img {
  align-items: center;
  width: 2.5rem;
  height: 2.5rem;
  margin-right: 0.5rem;
  border-radius: 50%;
}


.item_key {
  color: black;
}

.item_value {
  color: red;
  margin-left: 1rem;
}


@media only screen and (max-width: 800px) {
  .content_column {
    margin-right: 0;
    margin-top: 2rem;
    padding-left:2rem;
    width: 90%;
    padding-top:0;
    padding-bottom:0;
  }
  .item_key {
      margin-left: 1rem;
  }
  .item_text p {
    align-self: flex-start;
  }
  .item_text {
    flex-direction: column;
    align-items: flex-start;
    text-align: left;
  }
}
