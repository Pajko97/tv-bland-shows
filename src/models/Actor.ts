interface Actor {
  person: {
    id: number;
    url: string;
    name: string;
    country: {
      name: string;
      code: string;
      timezone: string;
    };
    birthday: string;
    deathday: null | any;
    gender: string;
    image: {
      medium: string;
      original: string;
    };
    updated: number;
    _links: {
      self: {
        href: string;
      };
    };
  };
  character: {
    id: number;
    url: string;
    name: string;
    image: {
      medium: string;
      original: string;
    };
    _links: {
      self: {
        href: string;
      };
    };
  };
  self: false;
  voice: false;
}
export default Actor;
