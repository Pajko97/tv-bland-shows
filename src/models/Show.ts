interface SingleShow {
    averageRuntime: number,
    dvdCountry: null
    ended: string,
    externals: Object
    genres: Array<any>,
    id: number,
    image: {
        medium : string,
        original : string
    },
    language: string
    name: string,
    network: {
        country : object,
        id: number,
        name : string
    },
    officialSite: null,
    premiered: string,
    rating: {
        average : any
    },
    runtime: number,
    schedule: {
        time : string,
        days: [string]
    },
    status: String,
    summary: string,
    type: string,
    updated: number,
    url: string,
    webChannel: null,
    weight: number,
    _links: Object
}   

export default SingleShow