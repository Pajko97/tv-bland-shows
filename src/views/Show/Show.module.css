.show_wrapper {
  padding: 10rem;
  font-family: "Red Hat Display";
}

.show_header {
  display: flex;
}
.show_header img {
  margin-right: 5rem;
  min-height: 100%;
  max-width: 25rem;
  max-height: auto;
}

.show_basic_info {
  max-width: 40rem;
}
.show_basic_info p {
  line-height: 150%;
}

.show_basic_info .rate {
  margin-left: 0.5rem;
}

.show_details {
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
}

@media only screen and (max-width: 800px) {
  .show_wrapper {
    padding: 3rem;
  }
  .show_basic_info {
    margin-top: 2rem;
  }
  .show_header {
    display: flex;
    flex-direction: column;
  }
  .show_header img {
    margin-right: 0;
  }
  .show_details {
    flex-direction: column;
  }
}
