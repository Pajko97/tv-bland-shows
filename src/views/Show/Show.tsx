import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router';
import API from '../../api/shows-api';
import SingleShow from '../../models/Show'
import striptags from '../../helpers/striptags'
import Rating from 'react-rating'
import ActorList from "../../components/Show/ActorList/ActorList"
import style from './Show.module.css'
import ShowInfo from '../../components/Show/ShowInfo/ShowInfo';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faCoffee, faStar, faStarHalfStroke } from '@fortawesome/free-solid-svg-icons'

type Props = Object

function Show(props: Props) {
  const [show, setShow] = useState<SingleShow>()

  let { id }: any = useParams();
  let showId = parseInt(id)

  useEffect(() => {
    let api = new API();
    api.getShow(showId)
      .then(res => setShow(res.data))
  }, [])

  const calcRating = (rating: number) =>
	rating > 5 ? Math.round((rating / 2) * 10) / 10 : rating

  return (
    <React.Fragment>
      {show && <div className={style.show_wrapper}>
        <div>
          <div className={style.show_header}>
            <img src={show.image.medium} alt="show cover"></img>
            <div className={style.show_basic_info}>
            <Rating
              emptySymbol={<FontAwesomeIcon icon={faStarHalfStroke} />}
              readonly
              initialRating={calcRating(show.rating.average)}
              fullSymbol={<FontAwesomeIcon icon={faStar}/>}

            />
            <span className={style.rate}>{show.rating.average ? `${show.rating.average}/5` : 'Not provided'}</span>
            <h1>{show.name}</h1>
            <p>{striptags(show.summary)}</p>
            </div>
          </div>

          <div className={style.show_details}>
            <ShowInfo show={show} />
            <ActorList showId={showId} />
          </div>
        </div>
      </div>}
    </React.Fragment>
  );
}

export default Show

