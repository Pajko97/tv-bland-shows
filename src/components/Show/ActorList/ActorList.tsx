import React, { useEffect, useState } from 'react';
import API from '../../../api/shows-api';
import Actor from '../../../models/Actor';
import style from './ActorList.module.css';

type ComponentProps = {
  showId: number
}


export default function ActorList(props: ComponentProps) {
  const [showCast, setShowCast] = useState<Array<Actor>>([])

  useEffect(() => {
    let api = new API();
    api.getShowCast(props.showId)
      .then(res => setShowCast(res.data))
  }, [])

  return (
    <div className={style.content_column}>
      <h2 className={style.item_headline}>Actors</h2>
      {showCast && showCast.map((actor, i) =>
        <div key={i} className={style.column_item}>
          <p className={style.item_key}><img src={actor.person.image.medium ? actor.person.image.medium : 'https://via.placeholder.com/30x30'} /></p>
          <div className={style.item_text}>
            <p className={style.item_key}>{actor.person.name}</p>
            <p className={style.item_value}>{actor.character.name}</p>
          </div>
        </div>
      )}
    </div>
  );
}
