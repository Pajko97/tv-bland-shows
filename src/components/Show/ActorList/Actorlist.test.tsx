import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import API from '../../../api/shows-api';
import Actor from '../../../models/Actor';
import Show from '../../../models/Show';
import ActorList from './ActorList';

it('render app', async () => {
    let render;
    let api = new API()
    await api.getShowCast(1)
    .then((res) => {
      render = renderer.create(
      <BrowserRouter>
        <ActorList showId={4}/>
      </BrowserRouter>).toJSON();
    })
    .catch((error) => console.error(error));

  expect(render).toMatchSnapshot();

})