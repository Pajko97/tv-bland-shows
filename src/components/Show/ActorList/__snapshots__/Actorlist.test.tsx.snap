// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders home 1`] = `
<div
  className="content_column"
>
  <h2
    className="item_headline"
  >
    Actors
  </h2>
</div>
`;
